src/ngraph/runtime/cpu/builder/pad.cpp

```diff
@@ -100,6 +100,69 @@ namespace ngraph
             }
 
             REGISTER_OP_BUILDER(Pad);
+
+            template <>
+            NodeExecutorTy Builder::BUILDER_CF_DECL(ngraph::op::Pad)
+            {
+                auto pad = static_cast<const ngraph::op::Pad*>(node);
+
+                auto arg_shape = pad->get_argument(0)->get_shape();
+                auto out_shape = pad->get_shape();
+                auto padding_below = pad->get_padding_below();
+                auto padding_above = pad->get_padding_above();
+
+                if (pad->get_padding_interior() == Shape(arg_shape.size()))
+                {
+                    std::function<decltype(runtime::cpu::kernel::pad<float, 1>)> kernel;
+
+                    SELECT_KERNEL_BY_RANK(kernel,
+                                          pad->get_input_element_type(0),
+                                          arg_shape.size(),
+                                          runtime::cpu::kernel::pad);
+
+                    auto functor = [kernel, arg_shape, out_shape, padding_below, padding_above](
+                        const std::vector<void*> inputs, std::vector<void*> outputs) {
```

Comment:
I see we are passing vector by value instead of by reference here and in many other places of this code. Is this intentional? Should we use references instead?